internal/util/concurrency/pool.go

```diff
@@ -16,24 +16,35 @@
 
 package concurrency
 
-import "github.com/panjf2000/ants/v2"
+import (
+	"runtime"
+
+	"github.com/panjf2000/ants/v2"
+)
 
 // A goroutine pool
 type Pool struct {
 	inner *ants.Pool
 }
 
-// Return error if provides invalid parameters
-// cap: the number of workers
-func NewPool(cap int, opts ...ants.Option) (*Pool, error) {
+// NewPool returns a goroutine pool.
+// cap: the number of workers.
+// This panic if provide any invalid option.
+func NewPool(cap int, opts ...ants.Option) *Pool {
 	pool, err := ants.NewPool(cap, opts...)
 	if err != nil {
-		return nil, err
+		panic(err)
```

Comment:
why use panic other than return error

Reply:
We never create a pool depending on the input from user, and only while init stage. This makes the constructor of `Pool` simpler so we can feel free to use it. Mostly, we don't know how to continue with error of creating pool.

In fact, this returns error only when we pass invalid arguments, I will later hide these arguments from ants lib